presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSink.java

```diff
@@ -442,11 +466,34 @@ public int[] partitionPage(Page page)
             for (int i = 0; i < columns.size(); i++) {
                 PartitionColumn column = columns.get(i);
                 Block block = page.getBlock(column.getSourceChannel());
+                // adjust legacy timestamp value to compatible with Iceberg non-identity transform calculation
+                if (column.sourceType instanceof TimestampType && session.getSqlFunctionProperties().isLegacyTimestamp() && !column.getField().transform().isIdentity()) {
+                    TimestampType timestampType = (TimestampType) column.sourceType;
+                    BlockBuilder bb = timestampType.createBlockBuilder(null, block.getPositionCount());
```

Comment:
nit: use full names instead of abbreviations

```suggestion
                    BlockBuilder blockBuilder = timestampType.createBlockBuilder(null, block.getPositionCount());
```